### PR TITLE
Extension Management Generalization

### PR DESCRIPTION
**Pull request by t6d**
### WHY are these changes introduced?

 This PR makes extension management more generic and future-proof.

### WHAT is this pull request doing?

* Removes surface area specific Argo Feature implementations in favour of a more configurable Argo implementation.
* Implements a default specification handler.
* Removes custom product subscription specification handler.
* Simplifies checkout post purchase specification handler.

### Update checklist

- [ ] ~~I've added a CHANGELOG entry for this PR (if the change is public-facing)~~
- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows).

### CODE REVIEW

File: lib/project_types/extension/features/argo.rb [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
```
Comment (t6d) [inline lib/project_types/extension/features/argo.rb:1]: This file has been moved and renamed. No other changes have been made.

File: lib/project_types/extension/features/argo.rb [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+require 'shopify_cli'
+require 'semantic/semantic'
+
+module Extension
+  module Features
+    class Argo
+      include SmartProperties
+
+      property! :git_template, converts: :to_str
+      property! :renderer_package_name, converts: :to_str
```
Comment (t6d) [inline lib/project_types/extension/features/argo.rb:12]: `git_template` and `renderer_package_name` are now injectable. The rest of the file is marked as new do to renaming and git not properly picking up on the file location change.

File: lib/project_types/extension/models/specification.rb [hunk 3]
```diff
@@ -1,8 +1,34 @@
...
+              .call(identifier, namespace: Features)
+              .rescue { OpenStruct }
+              .then { |c| c.new(**feature_attributes) }
+              .unwrap { |error| raise error }
+          end
+        end
+      end
+
+      property! :identifier
+      property :graphql_identifier, converts: :to_str
```
Comment (t6d) [inline lib/project_types/extension/models/specification.rb:27]: Could potentially be extracted into `messages` as this just influences CLI presentation and what `--type` takes as a value.

File: lib/project_types/extension/models/specifications.rb [hunk 4]
```diff
@@ -41,21 +45,86 @@ def fetch_specifications_and_build_handlers
...
-          handlers[handler.identifier] = handler
+          ShopifyCli::ResolveConstant.call(specification.identifier, namespace: custom_handler_namespace)
+            .rescue { |error| error.is_a?(NameError) ? SpecificationHandlers::Default : raise(error) }
+            .then { |handler_class| handler_class.new(specification) }
+            .unwrap { |error| raise error }
+            .yield_self { |handler| handlers[handler.identifier] = handler }
+        end
+      end
+
+      class ConfigureArgoFeature
```
Comment (t6d) [inline lib/project_types/extension/models/specifications.rb:73]: Rename to `ConfigureFeatures`

File: test/project_types/extension/features/argo_test.rb [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    class ArgoTest < MiniTest::Test
+      include TestHelpers::FakeUI
+      include ExtensionTestHelpers::Stubs::ArgoScript
+
+      def setup
+        super
+        ShopifyCli::ProjectType.load_type(:extension)
+
+        @dummy_argo = ExtensionTestHelpers::DummyArgo.new(
+          git_template: 'dummy-tempalte',
```
Comment (t6d) [inline test/project_types/extension/features/argo_test.rb:17]: Fix typo

**Review by fredriculrich — APPROVED** [review]
🚢

File: test/project_types/extension/features/argo_test.rb [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+      include ExtensionTestHelpers::Stubs::ArgoScript
+
+      def setup
+        super
+        ShopifyCli::ProjectType.load_type(:extension)
+
+        @dummy_argo = ExtensionTestHelpers::DummyArgo.new(
+          git_template: 'dummy-template',
+          renderer_package_name: 'dummy-renderer',
+        )
```
Comment (t6d) [inline test/project_types/extension/features/argo_test.rb:19]: This is the only change made in this file.

**Review by cecyc — APPROVED** [review]
🚀

**Comment by t6d** [comment]
🎩 completed!

I tested `create`, `register` and `push` for both extension types successfully.

**Pull request merged**